### ISSUE TITLE
Work Queues

### ISSUE THREAD

**Issue by Delaunay** [Epistimio/orion#359]
# Why 

Current system is rigid and does not give a lot of control into how work is distributed.
Nor is it easy to change or build on top of it.

Work queues standardize how work is distributed and how results are collected enabling us to make Orion db agnostic. This will allow us to leverage other software such as Dask to schedule and distribute the training of trials.

# Example

HPO master 
```python
queue_uri = 'mq://20.30.120.30:8123

# launch workers somewhere, could be on different nodes using SLURM job array
workers = [Worker(queue_uri) for i in range(10)]

# Launch HPO can be on another node, the users own computer etc...
hpo = HPO(queue_uri, space, function_to_optimize)
```
![hpo_is_independent](https://user-images.githubusercontent.com/4311937/76592330-6392f180-64c9-11ea-988e-1f584e3c9a56.png)

Nomad HPO: HPO is a simple task done by a worker that is instantiated when needed

```python
queue_uri = 'mq://20.30.120.30:8123

# tell a worker to start a HPO that will generate work items
put(queue_uri, 'HPO',  space, function_to_optimize)

# launch workers somewhere, could be on different nodes using SLURM job array
workers = [Worker(queue_uri) for i in range(10)]
```

![hpo_is_worker](https://user-images.githubusercontent.com/4311937/76592356-79a0b200-64c9-11ea-83a5-f3ec5a715fc5.png)

**Comment by bouthilx** [comment]
Agreed! Once we tackle #358, this refactoring of the workers into a worker queue should be much more simple.

**Comment by guillaume-chevalier** [comment]
I think you'll want to check this out: 
https://github.com/Neuraxio/Neuraxle/issues/221

**Comment by guillaume-chevalier** [comment]
@bouthilx @Delaunay we also have done a first pass of coding what I've referred to in the issue above: https://github.com/Neuraxio/Neuraxle/pull/242
